pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -382,13 +436,33 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 		return false, fmt.Errorf("failed to get cached claims to provision for pod %q", podName)
 	}
 
+	// Check for any conditions that might require scheduling retry
+
 	for _, binding := range bindings {
-		// Check for any conditions that might require scheduling retry
+		// Check against latest API object
+		obj, err := b.pvCache.GetAPIObj(binding.pv.Name)
```

Comment:
Maybe add a new method `GetApiPV` [here](https://github.com/kubernetes/kubernetes/blob/cdeb9205230ac6e5ea772ab8c562738d7c42bc3e/pkg/controller/volume/persistentvolume/scheduler_assume_cache.go#L341) to make the logic here easier to read.

Reply:
Done in 80d8815f64fb1452d8c62c2513707b90401d2b10.